src/mqttproto/client_state_machine.py

```diff
@@ -38,11 +38,8 @@ class MQTTClientStateMachine(BaseMQTTClientStateMachine):
     )
     _ping_pending: bool = field(init=False, default=False)
     _may_retain: bool = field(init=False, default=True)
+    _may_subscription_id: bool = field(init=False, default=True)
```

Comment:
I think this field name is a bit odd.
```suggestion
    _may_subscription_id: bool = field(init=False, default=True)
```